### PR TITLE
Add firewall v3 E2E tests

### PR DESCRIPTION
**Pull request by robertschaedler3**
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [x] All unit tests are passing.
- [x] I have merged the latest `main` branch prior to this PR submission.
- [x] I submitted this PR against the `main` branch.

### CODE REVIEW

File: src/tests/e2etest/FirewallTests.cs [hunk 1]
```diff
@@ -45,5 +115,91 @@ public void FirewallTest_Regex()
...
+            SetDesired<Rule[]>(_componentName, _desiredRules, desired);
+            GetReported<Firewall>(_componentName, (Firewall firewall) => (firewall.Fingerprint != initialFingerprint));
+
+            // Reset the desired state to empty
+            SetDesired<Rule[]>(_componentName, _desiredRules, new Rule[0]);
+
+            Assert.Multiple(() =>
+            {
+                Assert.AreEqual(State.Enabled, reported.State);
+                Assert.AreNotEqual(initialFingerprint, reported.Fingerprint);
```
Comment (simonjaeger) [inline src/tests/e2etest/FirewallTests.cs:164]: Would it make sense to also have assertions on the rules, in addition to the fingerprint?

Comment (robertschaedler3) [reply src/tests/e2etest/FirewallTests.cs:164]: Yes, however rules are not reported by the Firewall module only the fingerprint is reported as a representation of those rules.

File: src/tests/e2etest/FirewallTests.cs [hunk 2]
```diff
@@ -27,10 +30,77 @@ public enum State
...
+        [JsonConverter(typeof(StringEnumConverter), typeof(CamelCaseNamingStrategy))]
+        public enum Protocol
+        {
+            Any,
+            Tcp,
+            Udp,
+            Icmp
+        }
+
+        public partial class Rule
```
[the remark this answers is not in the document]
Comment (robertschaedler3) [reply src/tests/e2etest/FirewallTests.cs]: Yes good catch! Removed `partial` here and elsewhere.

**Comment by simonjaeger** [comment]
LGTM

**Review by MariusNi — APPROVED** [review]
:shipit:

**Pull request merged**